src/prefect/client/utilities.py

```diff
@@ -0,0 +1,62 @@
+import time
+from datetime import datetime
+
+import pendulum
+
+from prefect.client.orion import get_client, schemas
+from prefect.utilities.asyncutils import sync_compatible
+
+
+@sync_compatible
+async def run_deployment(
+    name: str,
+    parameters: dict = None,
+    scheduled_time: datetime = None,
+    max_polls: int = 60,
```

Comment:
In the past, I've preferred `timeout` or `timeout_seconds` where you can check the elapsed time at the start of each poll interval. This way, the user does not need to do math and it's clear that if you set a timeout you may get a timeout error.

It'd also be great to have it default to `-1` or `None` to to match other functions that can timeout (e.g. they'll run to completion unless otherwise specified). I could see the argument for `0` or returning immediately as well. Setting a default timeout doesn't feel as helpful though since it will not be clear on first-use to a user that it may timeout.

Reply:
Right, default of -1 makes sense -- if we poll, we should by default poll until completion 👍